ios/FluentUI/Drawer/DrawerController.swift

```diff
@@ -847,7 +866,13 @@ open class DrawerController: UIViewController {
                     updatePreferredContentSize(isExpanded: extraContentSizeEffect == .resize)
                 }
             }
-            presentationController.setExtraContentSize(offset)
+        
+            // When a valid 'preferredMaximumExpansionHeight' is provided, restrict the expansion animation to that height.
+            if((preferredMaximumExpansionHeight != -1 && originalDrawerHeight + offset <= preferredMaximumExpansionHeight) || (preferredMaximumExpansionHeight == -1))
```

Comment:
This logic should be added in the frameForContentView in the DrawerPresentationController.swift file.
The offset is calculated based on the gesture and it may cause jumps on the UI if the pan gesture is exaggerated.
I'll send you a video of the visual issue of jumps and bypassing max height with the gesture.

Reply:
moved this to DrawerPresentationController. Please have a look now.